### ISSUE TITLE
Issue with middleware execution multiple times

### ISSUE THREAD

**Issue by y-udaykiran96** [jeremydaly/lambda-api#75]
I'm using middleware for every request,
````
api.use('/*', (req, res, next) => {
    // do something
    req.log.info("from middleware",req.path);
    next();
  });
````
It's getting executed multiple times per request, logs are mentioned below
````
{"level":"info","time":1542865312652,"id":null,"route":"/locations","method":"DELETE","msg":"from middleware","timer":1,"custom":"/search/locations?abc=def"}
{"level":"info","time":1542865312652,"id":null,"route":"/locations","method":"DELETE","msg":"from middleware","timer":1,"custom":"/search/locations?abc=def"}
{"level":"info","time":1542865312652,"id":null,"route":"/locations","method":"DELETE","msg":"from middleware","timer":1,"custom":"/search/locations?abc=def"}
{"level":"info","time":1542865312652,"id":null,"route":"/locations","method":"DELETE","msg":{},"timer":1}
{"level":"info","time":1542865312652,"id":null,"route":"/locations","method":"DELETE","msg":"from locations any request","timer":1}
{"level":"access","time":1542865312653,"id":null,"route":"/locations","method":"DELETE","timer":2,"statusCode":203,"coldStart":false,"count":3,"path":"/search/locations?abc=def","ua":"PostmanRuntime/7.3.0","device":"unknown","country":"unknown","version":"v2"}
````
for 1st request, everything is fine, but for later requests, the middleware code is executing multiple times.
Please help me, if any code changes or fix this issue.

**Comment by jeremydaly** [comment]
Hi @y-udaykiran96! Thanks for reporting this. Seems like strange behavior, so I'll take a look.

**Comment by jeremydaly** [comment]
Thanks, @y-udaykiran96. I’ll take a look.

**Comment by jeremydaly** [comment]
@y-udaykiran96, are you placing the `api.use()` call inside of your main handler?

**Comment by y-udaykiran96** [comment]
@jeremydaly 
No, I'm not using lambda-api in main handler, 
I'm forwarding request object to router.js file, which contains lambda-api & all the rest-api segregations.

**Comment by jeremydaly** [comment]
@y-udaykiran96, can you send me the relevant parts of your code? I can't replicate this issue.

**Comment by jeremydaly** [comment]
@y-udaykiran96, can you see if v0.10 fixes the issue you were having? I've changed the way middleware processing works, so it shouldn't ever process more than once.

**Comment by jeremydaly** [comment]
Assuming this was fixed in v0.10 since I've not heard back and can't duplicate the issue. Closing for now.
